### ISSUE TITLE
JDK21 default DateTimeFormatter behind <f:convertDateTime type="localTime" /> doesn't anymore accept plain vanilla space

### ISSUE THREAD

**Issue by BalusC** [eclipse-ee4j/mojarra#5399]
Discovered in TCK https://github.com/jakartaee/faces/pull/1882

The following doesn't anymore work for me with OpenJDK 21.0.1:
```
<h:inputText id="localTime" value="#{backingBean.localTime}">
    <f:convertDateTime type="localTime" />
</h:inputText>
```
Assuming `en_US` locale, and no `pattern` specified, this creates under the covers a default formatter as follows:
```
DateTimeFormatter.ofLocalizedTime(FormatStyle.MEDIUM).withLocale(Locale.US);
```
which essentially defaults to a pattern `h:mm:ss a` however the whitespace before the `a` has somehow become a [NNBSP `\u202f`](https://www.fileformat.info/info/unicode/char/202f/index.htm) instead of a [plain vanilla space `\u0020`](https://www.fileformat.info/info/unicode/char/0020/index.htm). This means that the enduser has to literally type and submit the NNBSP character in order to get it to convert successfully. This makes no sense. I'm not terribly sure if the bug is in JDK or in Faces/Mojarra. But this is definitely something to look at. There should be a way to extract the pattern from the formatter and then replace all spurious whitespace by a normal space and then recreate the formatter based on `pattern` instead of `FormatStyle` as that appears to be intended as output-only somehow.

**Comment by BalusC** [comment]
On second thought, I believe defaulting to an arbitrary `FormatStyle` was a bad idea from the beginning on. It should in case of `javax.time` type have defaulted to ISO format, as defined in among others [`LocalTime#parse(String)`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/time/LocalTime.html#parse(java.lang.CharSequence)) (without formatter).

**Comment by chris21k** [comment]
Please see here for more information:
https://bugs.openjdk.org/browse/JDK-8324308
https://unicode-org.atlassian.net/browse/CLDR-17324

As far as I can understand, Unicode specification should allow different types of spaces.

Chris

**Comment by melloware** [comment]
@BalusC yes I got burned by this somewhere between JDK 11 and 17 when the Brazilian currency when from SPACE to NBSP.

I had to add this code to the CurrencyValidator in PrimeFaces...

https://github.com/primefaces/primefaces/blob/b96b824f895030c4f1b702c5e4918d37ac15f14b/primefaces/src/main/java/org/primefaces/util/CurrencyValidator.java#L142-L145
